pom.xml

```diff
@@ -6,7 +6,7 @@
   <parent>
     <groupId>org.apache</groupId>
     <artifactId>apache</artifactId>
-    <version>24</version>
+    <version>31</version>
```

Comment:
Is the upgrade of ASF pom parent mandatory for this PR? (I know that is good to upgrade but may be in another PR for the record and commit diff)

Reply:
These need to go together. Separately things will break